src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,259 @@
+        SmartDashboard.putNumber("InternalArmVelocity", armEncoder.getVelocity());
+        //SmartDashboard.putNumber("Arm Current", armMotor.getOutputCurrent());
+
+       // SmartDashboard.putNumber("ArmPos", getArmPos());
+
+
+
+        autoCancelArmCommand();
+    }
+
+    public TrapezoidProfile.State calculateCustomSetPoint(double goalSeconds, TrapezoidProfile.State currentPoint, TrapezoidProfile.State goalState) {
+        return armProfile.calculate(goalSeconds, currentPoint, goalState);
+        
+      }
+
+  public void autoCancelArmCommand() {
+        if(!(getDefaultCommand() instanceof ArmTeleop) || DriverStation.isAutonomous()) return;
+
+        double requestedSpeeds = ((ArmTeleop) getDefaultCommand()).getRequestedSpeeds();
+
+        if(requestedSpeeds != 0) {
+            Command currentArmCommand = getCurrentCommand();
+            if(currentArmCommand != getDefaultCommand() && currentArmCommand != null) {
+                currentArmCommand.cancel();
+            }
+        }
+    }
+
+    //#region Drive Methods
+    public void driveArm(double goalAngle){
+      TrapezoidProfile.State goalState = new TrapezoidProfile.State(goalAngle, 0);
+      TrapezoidProfile.State setPoint = armProfile.calculate(kDt, getCurrentArmState(), goalState);
+      double armFeedVolts = armFeed.calculate(goalState.velocity, 0);
+      if ((getArmPos() < ARM_LOWER_LIMIT_RAD && getCurrentArmGoal().velocity > 0) || (getArmPos() > ARM_UPPER_LIMIT_RAD && getCurrentArmGoal().velocity > 0)){
+        armFeedVolts = armFeed.calculate(getCurrentArmGoal().position, 0);
+      }
+      armPID1.setReference(setPoint.position, CANSparkBase.ControlType.kVelocity, 0, armFeedVolts);
+      armPID2.setReference(setPoint.position, CANSparkBase.ControlType.kVelocity, 0, armFeedVolts);
+    }
+
+    public void setArmTarget(double targetPos) {
+        targetPos = getArmClampedGoal(targetPos);
+
+        armProfile = new TrapezoidProfile(TRAP_CONSTRAINTS);
+       
 
+        goalState.position = targetPos;
+        goalState.velocity = 0;
     }
+
     
-    public void setSpeed() {
-      //move the arm around based off the right joystick movement on the manipulator joystick
-      //use the trapezoid thingy from robot code 2023
+
+    public void resetGoal() {
+        double armPos = getArmPos();
+      
+        armProfile = new TrapezoidProfile(TRAP_CONSTRAINTS);
+
+
+    }
+    public void driveMotor(Measure<Voltage> volts) {
+       armMotorMaster.setVoltage(volts.in(Volts));
+       armMotorFollower.setVoltage(volts.in(Volts));
+    }
+    public void logMotor(SysIdRoutineLog log) {
+        log.motor("armMotorMaster")
+                .voltage(voltage.mut_replace(
+                        armMotorMaster.getBusVoltage() * armMotorMaster.getAppliedOutput(),
+                        Volts))
+                .angularVelocity(velocity.mut_replace(
+                        armEncoder.getVelocity() / 60,
```

Comment:
be careful and make sure you are logging rotations per second. Be wary of the stuff you wrote in the constructor.